device/api/umd/device/coordinate_manager.h

```diff
@@ -149,19 +162,25 @@ class CoordinateManager {
      * should override this method. Blackhole coordinate manager overrides
      * this method to implement different mapping.
      */
-    virtual void fill_pcie_physical_translated_mapping();
+    virtual void fill_pcie_physical_translated_mapping() = 0;
 
     /*
      * Fills the physical to translated mapping for the ARC cores.
      * By default, translated coordinates are the same as physical coordinates.
      * Derived coordinate managers that need to implement different mapping
      * should override this method.
      */
-    virtual void fill_arc_physical_translated_mapping();
+    virtual void fill_arc_physical_translated_mapping() = 0;
 
     std::map<tt::umd::CoreCoord, tt_xy_pair> to_physical_map;
     std::map<std::pair<tt_xy_pair, CoordSystem>, tt::umd::CoreCoord> from_physical_map;
 
+    // Whether NOC translation is enabled.
+    // Effectivelly, when this is enabled, we can interface NOC with Virtual coordinates (on wormhole, translated by
```

Comment:
I would avoid mentioning virtual coordinates, since that is something that we came up with, lets say. We shouldn't use those coordinates for NOC endpoints, we should use either Physical (NOC0) or translated. Virtual are on API level at the moment